data/tests/tests_framework.txt

```diff
@@ -50,3 +50,181 @@ test "Test-Framework - Conditions Arithmetic and Loops"
+		input
+			key p
+		# Enter and leave shipyard
+		# Skipped, since Earth (the starting planet) doesn't have a shipyard.
+		#input
+		#	key s
+		#input
+		#	key l
+		# Enter and leave outfitter
+		input
+			key o
+		input
+			key l
+		# Enter hire-crew section
+		input
+			key h
+		# Depart
+		input
+			key d
+
+
+test "Test-Framework - UI Pointer controls"
+	status "missing feature"
+	description "Tests if the in-game panels can be controlled by pointer controls"
+	sequence
+		# Create/inject the savegame, go to the load screen and load the first (and only) savegame.
+		inject "Three Earthly Barges"
+		input
+			key l
+		input
+			key l
+		# Enter and leave jobs
+		input
+			key j
+		input
+			key d
+		# Enter trading
+		input
+			pointer
+				# Go from middle of the screen (to allow many resolutions)
+				X 0.5 relative
+				Y 0.5 relative
+				# Go for the center of the trading button
+				X -411
+				Y 92
+				click left
+		# Enter bank
+		input
+			pointer
+				# Go from middle of the screen (to allow many resolutions)
+				X 0.5 relative
+				Y 0.5 relative
+				# Go for the center of bank button
+				X -409
+				Y 211
+				click left
+		# Enter spaceport
+		input
+			pointer
+				# Go from middle of the screen (to allow many resolutions)
+				X 0.5 relative
+				Y 0.5 relative
+				# Go for the center of the spaceport button
+				X -410
+				Y 270
+				click left
+		# Enter and leave shipyard
+		#   Skipped, since Earth (the starting planet) doesn't have a shipyard.
+		# Enter and leave outfitter
+		input
+			pointer
+				# Go from middle of the screen (to allow many resolutions)
+				X 0.5 relative
+				Y 0.5 relative
+				# Go for the center of the outfitter button
+				X 284
+				Y 148
+				click left
+		input
+			pointer
+				# Leave outfitter, button is relative to lower right corner.
+				X -47
+				Y -27
+				click left
+		# Enter hire-crew section
+		input
+			pointer
+				X 291
+				Y 209
+				click left
+		# Depart
+		input
+			pointer
+				X 291
+				Y 269
+				click left
+
+
+test "Test-Framework - Load and depart"
+	status active
+	description "Test with only a departure. Mostly tests the testframework itself."
+	sequence
+		# Create/inject the savegame, go to the load screen and load the first (and only) savegame.
+		inject "Three Earthly Barges"
+		input
+			key l
+		input
+			key l
+		# Check startup conditions.
+		assert
+			"ships: Light Freighter" = 3
+		# Launch.
+		input
+			key d
+		# Re-check startup conditions.
+		assert
+			"ships: Light Freighter" = 3
+
+
+test "Test-Framework - Navigate"
+	status active
+	description "Test-Framework - Navigation to other star system"
+	sequence
+		# Create/inject the savegame, go to the load screen and load the first (and only) savegame.
+		inject "Three Earthly Barges"
+		input
+			key l
+		input
+			key l
+		# Set desired travel plan.
+		navigate
+			travel "Alpha Centauri"
+			travel "Sol"
+			travel "Alpha Centauri"
+		# Launch.
+		input
+			key d
+		input
+		input
```

Comment:
What do empty inputs signify?

Reply:
I was debugging something here and expected that the game needed more than 1 step to perform the complete launch.
The test-loops return control for 1 game-step after an input statement or when encountering a branch for a second time. An input statement can be used to let the game to do a step.

Reply:
I just checked, the game does indeed need more than 1 step to perform the launch (in the current code-base).

Reply:
Added a comment to explain this better.